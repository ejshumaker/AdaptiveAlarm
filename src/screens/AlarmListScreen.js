import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  View, Text, FlatList, ActivityIndicator, TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';

import { GlobalStyles, Colors } from '../constants';

import { AlarmItem, StatusBarBackground } from '../components';
import { CloseIcon } from '../icons/close';

import { userSetAlarmStatus } from '../store/actions/userActions';


class AlarmListScreen extends Component {
  constructor() {
    super();
    this.state = {
    };
  }

  // Loading icon !
  loader() {
    const { loading } = this.props;
    if (loading) {
      return <ActivityIndicator color={Colors.primary} size="large" />;
    }
    return null;
  }

  menu() {
    const {
      navigation,
    } = this.props;
    const { navigate } = navigation;
    return (
      <View style={GlobalStyles.menu}>
        <TouchableOpacity
          onPress={() => { navigate('Main'); }}
        >
          <CloseIcon />
        </TouchableOpacity>
      </View>
    );
  }


  render() {
    const { alarms, toggleAlarm, navigation } = this.props;
    const { navigate } = navigation;

    return (
      <View style={{ flex: 1 }}>
        <StatusBarBackground />
        {this.menu()}
        <View style={[GlobalStyles.container, { paddingHorizontal: 48 }]}>
          <Text
            style={[
              GlobalStyles.h2,
              {
                color: Colors.primary,
                marginBottom: 48,
              },
            ]}
          >
          ALL ALARMS:
          </Text>
          {this.loader()}

          <FlatList
            showsHorizontalScrollIndicator={false}
            showsVerticalScrollIndicator={false}
            data={Object.values(alarms)}
            renderItem={({ item, index }) => (
              <AlarmItem
                alarm={item}
                alarmId={Object.keys(alarms)[index]}
                toggleAlarm={toggleAlarm}
                navigate={navigate}
              />
            )}
            keyExtractor={(item, index) => String(index)}
          />

        </View>
      </View>
    );
  }
  // end render
}
// end class

// I think this is almost completed ?
AlarmListScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  // Redux state
  loading: PropTypes.bool.isRequired,
  alarms: PropTypes.object, // eslint-disable-line
  // Redux dispatch
  toggleAlarm: PropTypes.func.isRequired,
};

AlarmListScreen.defaultProps = {
  alarms: {},
};

/**
 * @weinoh 04-10-19
 */
const mapStateToProps = state => ({
  alarmTime: state.alarm.time,
  alarmCalculating: state.alarm.loading,
  loading: state.user.loadingFetch,
  alarms: state.user.alarms,
});

/**
 * TODO: Does this toggleAlarm function need to be passed into our alarmitem?
 * @weinoh 04-10-19
 */
const mapDispatchToProps = dispatch => ({
  toggleAlarm: (alarmid, status) => {
    dispatch(userSetAlarmStatus(alarmid, status));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AlarmListScreen);
