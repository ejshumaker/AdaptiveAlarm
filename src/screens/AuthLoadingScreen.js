import React from 'react';
import {
  ActivityIndicator,
  View,
} from 'react-native';
import { connect } from 'react-redux';
import { auth } from 'firebase';
import PropTypes from 'prop-types';

import { userFetch } from '../store/actions/userActions';
import { GlobalStyles, Colors } from '../constants';

class AuthLoadingScreen extends React.Component {
  constructor(props) {
    super(props);
    this.bootstrapAsync();
  }

  // Fetch the token from storage then navigate to our appropriate place
  bootstrapAsync = async () => {
    const { navigation, fetchUser } = this.props;

    // This will switch to the App screen or Auth screen and this loading
    // screen will be unmounted and thrown away.
    auth().onAuthStateChanged((user) => {
      if (user) {
        // User is signed in.
        fetchUser(user.uid, navigation.navigate);
        navigation.navigate('App');
      } else {
        // User is signed out.
        navigation.navigate('Auth');
      }
    });
  };

  // Render simple spinner,
  // this is usually so fast you won't see it
  render() {
    return (
      <View style={GlobalStyles.centerChildrenXY}>
        <ActivityIndicator color={Colors.primary} size="large" />
      </View>
    );
  }
}

AuthLoadingScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  // Redux dispatch
  fetchUser: PropTypes.func.isRequired,
};

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => ({
  fetchUser: (user, navigate) => { dispatch(userFetch(user, navigate)); },
});

export { AuthLoadingScreen };
export default connect(mapStateToProps, mapDispatchToProps)(AuthLoadingScreen);
