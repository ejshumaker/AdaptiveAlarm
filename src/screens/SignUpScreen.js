import React from "react";
import {
  TextInput,
  View,
  ActivityIndicator,
  Text,
  ScrollView,
  SafeAreaView,
  StatusBar
} from "react-native";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { userCreateAccount } from "../store/actions/userActions";
import { Colors, GlobalStyles } from "../constants";
import Buttons from "../components/Buttons";
import { EmailIcon } from "../icons/email";
import { KeyIcon } from "../icons/key";
import { PersonIcon } from "../icons/person";
import { Number1Icon } from "../icons/one";
import { Number2Icon } from "../icons/two";

class SignUpScreen extends React.Component {
  state = {
    firstName: "",
    lastName: "",
    userName: "",
    password: "",
    email: ""
  };

  onChangeText(key, value) {
    this.setState({
      [key]: value
    });
  }

  loader() {
    const { loading } = this.props;
    if (loading) {
      return <ActivityIndicator color={Colors.primary} size="large" />;
    }
    return null;
  }

  render() {
    const { createAccount, navigation, errorMessage } = this.props;
    return (
      <View style={[GlobalStyles.container, { justifyContent: 'center', padding: 48 }]}>
        <StatusBar barStyle="light-content" />
        <Text
          style={[
            GlobalStyles.h2,
            { textAlign: "left", color: Colors.primary }
          ]}
        >
          SIGN UP:
        </Text>
        {this.loader()}
        <Text
          style={[
            GlobalStyles.paragraph,
            { color: Colors.error, marginVertical: 24 }
          ]}
        >
          {errorMessage}
        </Text>
        <View
          style={{
            flexDirection: "row",

            height: 40,
            backgroundColor: Colors.darkGray,
            borderRadius: 8,
            marginBottom: 30
          }}
        >
          <EmailIcon style={{ marginLeft: 13, marginTop: 7 }} />
          <TextInput
            keyboardAppearance={'dark'}
            onChangeText={value => this.onChangeText("email", value)}
            style={GlobalStyles.destinationInput}
            placeholder="Email"
            placeholderTextColor={Colors.gray}
          />
        </View>
        <View
          style={{
            flexDirection: "row",

            height: 40,
            backgroundColor: Colors.darkGray,
            borderRadius: 8,
            marginBottom: 30
          }}
        >
          <PersonIcon style={{ marginLeft: 13, marginTop: 7 }} />
          <TextInput
            keyboardAppearance={'dark'}
            onChangeText={value => this.onChangeText("userName", value)}
            style={GlobalStyles.destinationInput}
            placeholder="Username"
            placeholderTextColor={Colors.gray}
          />
        </View>
        <View
          style={{
            flexDirection: "row",

            height: 40,
            backgroundColor: Colors.darkGray,
            borderRadius: 8,
            marginBottom: 30
          }}
        >
          <KeyIcon style={{ marginLeft: 13, marginTop: 7 }} />
          <TextInput
            keyboardAppearance={'dark'}
            onChangeText={value => this.onChangeText("password", value)}
            style={GlobalStyles.destinationInput}
            secureTextEntry
            placeholder="Password"
            placeholderTextColor={Colors.gray}
          />
        </View>
        <View
          style={{
            flexDirection: "row",

            height: 40,
            backgroundColor: Colors.darkGray,
            borderRadius: 8,
            marginBottom: 30
          }}
        >
          <Number1Icon style={{ marginLeft: 13, marginTop: 7 }} />
          <TextInput
            keyboardAppearance={'dark'}
            onChangeText={value => this.onChangeText("firstName", value)}
            style={GlobalStyles.destinationInput}
            placeholder="First Name"
            placeholderTextColor={Colors.gray}
          />
        </View>
        <View
          style={{
            flexDirection: "row",

            height: 40,
            backgroundColor: Colors.darkGray,
            borderRadius: 8
          }}
        >
          <Number2Icon style={{ marginLeft: 13, marginTop: 7 }} />
          <TextInput
            keyboardAppearance={'dark'}
            onChangeText={value => this.onChangeText("lastName", value)}
            style={GlobalStyles.destinationInput}
            placeholder="Last Name"
            placeholderTextColor={Colors.gray}
          />
        </View>
        <View style={{ marginTop: 60, alignItems: "center" }}>
          <Buttons
            title="Sign Up"
            backgroundColor={Colors.primary}
            textColor={Colors.black}
            onPress={() => {
              const {
                firstName,
                lastName,
                email,
                userName,
                password
              } = this.state;
              createAccount({
                firstName,
                lastName,
                email,
                userName,
                password
              });
            }}
          />
          <Buttons
            title="Sign In"
            backgroundColor={Colors.darkGray}
            textColor={Colors.white}
            onPress={() => {
              navigation.navigate("SignIn");
            }}
          />
        </View>
      </View>
    );
  }
}

SignUpScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired,
  // Redux state
  errorMessage: PropTypes.string,
  loading: PropTypes.bool.isRequired,
  // Redux dispatch
  createAccount: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  errorMessage: state.user.errorMessage,
  loading: state.user.loading
});

SignUpScreen.defaultProps = {
  errorMessage: ""
};

const mapDispatchToProps = dispatch => ({
  createAccount: credentials => {
    dispatch(userCreateAccount(credentials));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignUpScreen);
