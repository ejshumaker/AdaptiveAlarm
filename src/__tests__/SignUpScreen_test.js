
/* eslint-disable no-undef */
/* eslint-disable import/first */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SignUpScreen } from '../screens/SignUpScreen';

jest.mock('firebase', () => ({
  auth: () => ({
    onAuthStateChanged: jest.fn(cb => cb({ name: 'Tristan', uid: 1 })),
  }),
}));

Enzyme.configure({ adapter: new Adapter() });


describe('SignUp Screen', () => {
  let wrapper;
  // our mock login function to replace the one provided by mapDispatchToProps
  const mockSignUpfn = jest.fn();
  const navigation = { navigate: jest.fn() };

  beforeEach(() => {
    // pass the mock function as the login prop
    wrapper = shallow(<SignUpScreen
      navigation={navigation}
      createAccount={mockSignUpfn}
      loading={false}
      // Redux state
      errorMessage=""
    />);
  });

  it('test SignUp screen matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('test SignUp screen matches snapshot after state set', () => {
    wrapper.setState({
      firstName: 'Tristan',
      lastName: 'Steiner',
      userName: 'tsteiner4',
      password: 'testPass',
      email: 'tsteiner4@wisc.edu',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the sign up action', () => {
    wrapper.setState({
      firstName: 'Tristan',
      lastName: 'Steiner',
      userName: 'tsteiner4',
      password: 'testPass',
      email: 'tsteiner4@wisc.edu',
    });
    wrapper.find('[title="Sign Up"]').simulate(
      'press',
      { preventDefault() {} },
    );
    expect(mockSignUpfn.mock.calls.length).toBe(1);
  });
});
