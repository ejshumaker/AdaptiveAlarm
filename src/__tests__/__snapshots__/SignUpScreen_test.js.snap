// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Screen test SignUp screen matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
        "padding": 48,
      },
    ]
  }
>
  <StatusBar
    barStyle="light-content"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Black",
          "fontSize": 30,
        },
        Object {
          "color": "#E4F96A",
          "textAlign": "left",
        },
      ]
    }
  >
    SIGN UP:
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 18,
        },
        Object {
          "color": "#fa0505",
          "marginVertical": 24,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <EmailIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <PersonIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Username"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <KeyIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#AAAAAA"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <Number1Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="First Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
      }
    }
  >
    <Number2Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 60,
      }
    }
  >
    <Buttons
      backgroundColor="#E4F96A"
      onPress={[Function]}
      textColor="#13151A"
      title="Sign Up"
    />
    <Buttons
      backgroundColor="#505050"
      onPress={[Function]}
      textColor="#DDDDDD"
      title="Sign In"
    />
  </View>
</View>
`;

exports[`SignUp Screen test SignUp screen matches snapshot after state set 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
        "padding": 48,
      },
    ]
  }
>
  <StatusBar
    barStyle="light-content"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Black",
          "fontSize": 30,
        },
        Object {
          "color": "#E4F96A",
          "textAlign": "left",
        },
      ]
    }
  >
    SIGN UP:
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 18,
        },
        Object {
          "color": "#fa0505",
          "marginVertical": 24,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <EmailIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <PersonIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Username"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <KeyIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#AAAAAA"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <Number1Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="First Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
      }
    }
  >
    <Number2Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 60,
      }
    }
  >
    <Buttons
      backgroundColor="#E4F96A"
      onPress={[Function]}
      textColor="#13151A"
      title="Sign Up"
    />
    <Buttons
      backgroundColor="#505050"
      onPress={[Function]}
      textColor="#DDDDDD"
      title="Sign In"
    />
  </View>
</View>
`;

exports[`SignUp Screen test SignUp screen matches snapshot loading 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
        "padding": 48,
      },
    ]
  }
>
  <StatusBar
    barStyle="light-content"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Black",
          "fontSize": 30,
        },
        Object {
          "color": "#E4F96A",
          "textAlign": "left",
        },
      ]
    }
  >
    SIGN UP:
  </Text>
  <ActivityIndicator
    color="#E4F96A"
    size="large"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 18,
        },
        Object {
          "color": "#fa0505",
          "marginVertical": 24,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <EmailIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <PersonIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Username"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <KeyIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#AAAAAA"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <Number1Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="First Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
      }
    }
  >
    <Number2Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 60,
      }
    }
  >
    <Buttons
      backgroundColor="#E4F96A"
      onPress={[Function]}
      textColor="#13151A"
      title="Sign Up"
    />
    <Buttons
      backgroundColor="#505050"
      onPress={[Function]}
      textColor="#DDDDDD"
      title="Sign In"
    />
  </View>
</View>
`;

exports[`SignUp Screen test SignUp screen matches snapshot not loading 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
        "padding": 48,
      },
    ]
  }
>
  <StatusBar
    barStyle="light-content"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Black",
          "fontSize": 30,
        },
        Object {
          "color": "#E4F96A",
          "textAlign": "left",
        },
      ]
    }
  >
    SIGN UP:
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 18,
        },
        Object {
          "color": "#fa0505",
          "marginVertical": 24,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <EmailIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <PersonIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Username"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <KeyIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#AAAAAA"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <Number1Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="First Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
      }
    }
  >
    <Number2Icon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      keyboardAppearance="dark"
      onChangeText={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#AAAAAA",
          "fontFamily": "RNSMiles-Medium",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 60,
      }
    }
  >
    <Buttons
      backgroundColor="#E4F96A"
      onPress={[Function]}
      textColor="#13151A"
      title="Sign Up"
    />
    <Buttons
      backgroundColor="#505050"
      onPress={[Function]}
      textColor="#DDDDDD"
      title="Sign In"
    />
  </View>
</View>
`;
